conjur/cli.py

```diff
@@ -16,31 +16,96 @@
 
 from .version import __version__
 
+class ArgparseWrapper(argparse.ArgumentParser):
+    def error(self, message):
+        sys.stderr.write('Error %s\n' % message)
+        self.print_help()
+        sys.exit(1)
+
+    def subcommand_error(self, message, help):
+        sys.stderr.write('Error %s\n' % message)
+        sys.stderr.write('%s\n' % help)
+        sys.exit(1)
+
+    def parse_args(self, args=None, namespace=None):
+        args, argv = self.parse_known_args(args, namespace)
+        action = args.resource if args else None
+        if argv:
+            msg = 'unrecognized arguments: %s'
+            err_msg = (msg % ' '.join(argv))
+            if action:
+                # Find the requesting action
+                action_help = self._subparsers._actions[0].choices.get(action)
+                if action_help:
+                    # Return the help for the specific action
+                    self.subcommand_error(err_msg, action_help.format_help())
+                else:
+                    self.error(err_msg)
+            else:
+                self.error(err_msg)
+        return args
 
 class Cli():
     """
     Main wrapper around CLI-like usages of this module. Provides various
     helpers around parsing of parameters and running client commands.
     """
 
+    @staticmethod
+    def main_description():
+        """
+        This method builds the header for the main screen
+        """
+        return '''
+        
+Conjur’s CLI for managing roles, resources and privileges
+
+Usage:
+    conjur [global options] command [subcommand] [options] [arguments…]'''
+
+    @staticmethod
+    def main_epilog():
+        """
+        This method builds the footer for the main help screen
+        """
+        return '''
+Copyright:
+    Copyright © 1999-2020 CyberArk Software Ltd. All rights reserved.
+    <www.cyberark.com>
+
+    To get help on a specific command, type: conjur <command> -h'''
```

Comment:
Because in Python, indentation is very important, you should compose these hanging lines using string addition to avoid visually confusing lines like these where they seem like they are part of the code or even the wrong nesting level.

Reply:
Are you referencing just the `conjur <command> -h` ? Or split this whole line up like so:
```
pre = "To get help on a specific command, type"
post = "conjur <command> -h"

pre + post
```